### PR TITLE
Make torch-harmonic an optional dependency

### PR DESCRIPTION
**Pull request by dhpitt**

### CODE REVIEW

File: neuralop/models/fno.py [hunk 1]
```diff
@@ -4,7 +4,13 @@
 import torch.nn.functional as F
 
 from ..layers.spectral_convolution import SpectralConv
-from ..layers.spherical_convolution import SphericalConv
+# Only reference SphericalConv if torch_harmonics is built locally
```
Comment (JeanKossaifi) [inline neuralop/models/fno.py:7]: What about just moving the whole sfno logic to an sfno file?

Comment (dhpitt) [reply neuralop/models/fno.py]: Sounds good. done

File: neuralop/models/sfno.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'''
```
Comment (JeanKossaifi) [inline neuralop/models/sfno.py:1]: Nitpick for consistency: We use """ generally in the library

File: neuralop/models/__init__.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 from .fno import TFNO, TFNO1d, TFNO2d, TFNO3d
 from .fno import FNO, FNO1d, FNO2d, FNO3d
-from .fno import SFNO
+# only import SFNO if torch_harmonics is built locally
+from importlib.util import find_spec
```
Comment (JeanKossaifi) [inline neuralop/models/__init__.py:4]: Following up on the other PR: let's use try/except everywhere if choose that

**Pull request merged**